modules/fortigate-firewall/variables.tf

```diff
@@ -0,0 +1,226 @@
+  }
+}
+
+variable "platform" {
+  description = "Device platform flavor that determines number of CPU cores and memory"
+  type        = string
+  validation {
+    condition     = can(regex("^(small|medium|large|xlarge)$", var.platform))
+    error_message = "One of following platform flavors are supported: small, medium, large, xlarge."
+  }
+}
+
+variable "software_package" {
+  description = "Device software package"
+  type        = string
+  validation {
+    condition     = can(regex("^(VM02|VM04|VM08|VM16)$", var.software_package))
+    error_message = "One of following software packages are supported: VM02, VM04, VM08, VM16."
+  }
+}
+
+variable "version_number" {
+  description = "Device software package version number"
+  type        = string
+  default     = ""
+}
+
+variable "license_file" {
+  description = "Path to the device license file"
+  type        = string
+  default     = ""
+}
+
+variable "license_token" {
+  description = "License token"
+  type        = string
+  default     = ""
+}
+
+variable "name" {
+  description = "Device name"
+  type        = string
+  validation {
+    condition     = length(var.name) >= 3 && length(var.name) <= 50
+    error_message = "Device name should consist of 3 to 50 characters."
+  }
+}
+
+variable "hostname" {
+  description = "Device hostname prefix"
+  type        = string
+  default     = null
+  validation {
+    condition     = try(var.cluster.enabled, false) || try(length(var.hostname) >= 2 && length(var.hostname) <= 30, false)
+    error_message = "Device hostname should consist of 2 to 30 characters."
+  }
+}
+
+variable "term_length" {
+  description = "Term length in months"
+  type        = number
+  validation {
+    condition     = can(regex("^(1|12|24|36)$", var.term_length))
+    error_message = "One of following term lengths are available: 1, 12, 24, 36 months."
+  }
+}
+
+variable "notifications" {
+  description = "List of email addresses that will receive device status notifications"
+  type        = list(string)
+  validation {
+    condition     = length(var.notifications) > 0
+    error_message = "Notification list cannot be empty."
+  }
+}
+
+variable "acl_template_id" {
+  description = "Identifier of an ACL template that will be applied on a device"
+  type        = string
+  validation {
+    condition     = try(length(var.acl_template_id) > 0, false)
+    error_message = "Acl template is required."
+  }
+}
+
+variable "additional_bandwidth" {
+  description = "Additional internet bandwidth for a device"
+  type        = number
+  default     = 0
+  validation {
+    condition     = var.additional_bandwidth == 0 || (var.additional_bandwidth >= 25 && var.additional_bandwidth <= 5001)
+    error_message = "Additional internet bandwidth should be between 25 and 5001 Mbps."
+  }
+}
+
+variable "ssh_key" {
+  description = "SSH public key for a device"
+  type        = object({
+    username = string
+    key_name = optional(string)
```

Comment:
any reason to declare key_name as optional attribute?

Reply:
Resolved.